extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,272 @@
+        {
+            bindingDataContent = bindingData.ContentType switch
+            {
+                Constants.JsonContentType => new Dictionary<string, string>(bindingData.Content.ToObjectFromJson<Dictionary<string, string>>(), StringComparer.OrdinalIgnoreCase),
+                _ => null
+            };
+
+            return bindingDataContent is not null;
+        }
+
+        private async Task<object?> ConvertModelBindingDataAsync(IDictionary<string, string> content, Type targetType, ModelBindingData bindingData)
+        {
+            content.TryGetValue(Constants.Connection, out var connectionName);
+            content.TryGetValue(Constants.ContainerName, out var containerName);
+            content.TryGetValue(Constants.BlobName, out var blobName);
+
+            if (string.IsNullOrEmpty(connectionName) || string.IsNullOrEmpty(containerName))
+            {
+                throw new ArgumentNullException("'Connection' or 'ContainerName' cannot be null or empty");
+            }
+
+            var result = await ToTargetType(targetType, connectionName, containerName, blobName);
+            return result;
+        }
+
+        private async Task<object?> ToTargetType(Type targetType, string connectionName, string containerName, string blobName) => targetType switch
+        {
+            Type _ when targetType == typeof(String) => await GetBlobString(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(Stream) => await GetBlobStream(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(Byte[]) => await GetBlobBinaryData(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(BlobBaseClient) => CreateBlobClient<BlobBaseClient>(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(BlobClient) => CreateBlobClient<BlobClient>(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(BlockBlobClient) => CreateBlobClient<BlockBlobClient>(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(PageBlobClient) => CreateBlobClient<PageBlobClient>(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(AppendBlobClient) => CreateBlobClient<AppendBlobClient>(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(BlobContainerClient) => CreateBlobContainerClient(connectionName, containerName),
+            _ => await DeserializeToTargetObject(targetType, connectionName, containerName, blobName)
+        };
+
+        private IEnumerable<object>? ToTargetTypeCollection(Type targetType, IEnumerable<object> blobCollection) => targetType switch
+        {
+            Type _ when targetType == typeof(IEnumerable<BlobBaseClient>) => blobCollection.Select(b => (BlobBaseClient)b),
```

Comment:
Just to be clear - we expect `targetType` to only ever be `IEnumerable<T>` here right? Not something like `ICollection<T>`?

Reply:
Yup, as far as I'm aware, that's all we advertise as what we support - mostly going off the sdk docs: https://github.com/surgupta-msft/azure-sdk-for-net/tree/main/sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs